### PR TITLE
Add pixeltris's TwitchAdSolutions scripts

### PR DESCRIPTION
**Pull request by ryanbr**
Include an option for users to filter out twitch.tv ads. 

2 different scripts, filtering in [2 different ways](https://github.com/pixeltris/TwitchAdSolutions?tab=readme-ov-file#scripts).

https://github.com/pixeltris/TwitchAdSolutions

The 2 scripts imported here:

https://github.com/pixeltris/TwitchAdSolutions/blob/master/video-swap-new/video-swap-new-ublock-origin.js
https://github.com/pixeltris/TwitchAdSolutions/blob/master/vaft/vaft-ublock-origin.js

Not global scripts, but will only apply too twitch.tv domain.

- `twitch.tv##+js(video-swap-new-ublock-origin)`
- `twitch.tv##+js(vaft-ublock-origin)`

Won't be a default, enabled option. But something we test via a seperate list or via experimental list.

### CODE REVIEW

**Comment by TEMP-ad** [comment]
I use the scriptlets on my side, and they seem to work okay, I use VAFT 99% of the time, unless it doesn't work I try Swap one. but it seems like in more than a year I have used them, everything works okay. I wonder if regions with more ads will have issues with these scriptlets.

But they always work, I disable them and get ads and then enable and ads are gone.

I don't know if you know it, but be aware that you can't add both rules at the same time because the Twitch video will not load if you have both rules loaded at the same time. That means you either have to choose which one you want people to use, which I think VAFT is the better one and the one I use, or have them in two separated lists, which would make things more complicated because now you need 3 experimental lists rather than just one. 

In my case I use the rules:

```
www.twitch.tv,embed.twitch.tv,player.twitch.tv##+js(twitch-videoad-vaft)
!www.twitch.tv,embed.twitch.tv,player.twitch.tv##+js(twitch-videoad-swap)
```
and sometimes I get black screen in one, and I have to test the other one, and it can be hit and miss, BUT I haven't had ads in twitch for a year, unless some updates in the resources file happen and I load a video and need to modify the resources file again with my scriptlets and these ones.

This would be good even if people have to manually add the rules, which could be another choice, especially since scriptlets injection exceptions can happen in the same Adblock engine, so it's wouldn't be hard for people to know the rules to use or something, and let them choose what experience they want ads or 360p while ads are going.

**Comment by ShivanKaul** [comment]
@bcaller can you take a look at these scriptlets? Plan would be to bundle them with the other scriptlets but only insert them on Twitch on an experimental basis (e.g. put them in the Experimental filter list).

**Comment by ryanbr** [comment]
Re-sync with latest changes in https://github.com/pixeltris/TwitchAdSolutions/commit/044d1fb3bb88a19ab718603069354d43eeba7015

File: resources/vaft-ublock-origin.js [hunk 1]
```diff
@@ -0,0 +1,917 @@
...
+    } catch (err) {}
+    //Send settings updates to worker.
+    window.addEventListener('message', (event) => {
+        if (event.source != window) {
+            return;
+        }
+        if (event.data.type && event.data.type == 'SetTwitchAdblockSettings' && event.data.settings) {
+            TwitchAdblockSettings = event.data.settings;
+        }
+    }, false);
```
[the remark this answers is not in the document]
Comment (bcaller) [reply resources/vaft-ublock-origin.js:53]: Where does this event come from?

File: resources/vaft-ublock-origin.js [hunk 2]
```diff
@@ -0,0 +1,865 @@
...
+                ${processM3U8.toString()}
+                ${hookWorkerFetch.toString()}
+                ${declareOptions.toString()}
+                ${getAccessToken.toString()}
+                ${gqlRequest.toString()}
+                ${adRecordgqlPacket.toString()}
+                ${tryNotifyTwitch.toString()}
+                ${parseAttributes.toString()}
+                declareOptions(self);
+                self.TwitchAdblockSettings = ${JSON.stringify(TwitchAdblockSettings)};
```
Comment (bcaller) [inline resources/vaft-ublock-origin.js:104]: Eeek, this is scary

Comment (bcaller) [reply resources/vaft-ublock-origin.js:104]: Oh, I see the first 11 are just printing the content of functions. Line 104 is printing the JSON which might have been problematic but we aren't in an HTML context.

**Review by bcaller — APPROVED** [review]
Looks OK

**Comment by bcaller** [comment]
Still curious where the `SetTwitchAdblockSettings` is supposedly fired from.

**Comment by ryanbr** [comment]
cc: @pixeltris

**Comment by pixeltris** [comment]
> Still curious where the `SetTwitchAdblockSettings` is supposedly fired from.

That `vaft` script was originally an extension and has stuff left over from that. It sounds like that code can be removed. The invoker was https://github.com/cleanlock/VideoAdBlockForTwitch/blob/master/chrome/content.js

File: resources/vaft-ublock-origin.js (hunk 2 above)
Comment (pixeltris) [reply resources/vaft-ublock-origin.js:104]: Removed `TwitchAdblockSettings` and related code in https://github.com/pixeltris/TwitchAdSolutions/commit/02447140c7522721601ac62c4fb47c7757e054d7 as it wasn't used.

And yes the weird looking code above is injecting code into their `Worker` to intercept `fetch` to modify the result of m3u8 requests.

**Review by antonok-edm — APPROVED** [review]

**Comment by ShivanKaul** [comment]
This LGMT but @bcaller could you take a quick look again? The scripts changed a bit (for the better I think).

**Review by ShivanKaul — APPROVED** [review]

**Review by bcaller — APPROVED** [review]

**Pull request merged**